scripts/pre_push_hook.py

```diff
@@ -174,32 +210,22 @@ def _collect_files_being_pushed(ref_list, remote):
     # Get branch name from e.g. local_ref='refs/heads/lint_hook'.
     branches = [ref.local_ref.split('/')[-1] for ref in ref_heads_only]
     hashes = [ref.local_sha1 for ref in ref_heads_only]
-    remote_hashes = [ref.remote_sha1 for ref in ref_heads_only]
     collected_files = {}
     # Git allows that multiple branches get pushed simultaneously with the "all"
     # flag. Therefore we need to loop over the ref_list provided.
-    for branch, sha1, remote_sha1 in zip(branches, hashes, remote_hashes):
-        # Git reports the following for an empty / non existing branch
-        # sha1: '0000000000000000000000000000000000000000'.
-        if set(remote_sha1) != {'0'}:
```

Comment:
Why don't we require this part, that is, the removed portion of the code now?

Reply:
Here it was comparing the changes of the local branch with the local branch itself if the branch is not empty. Now, we are making it compare to upstream at all time.